pkg/proxy/proxier.go

```diff
@@ -565,12 +573,35 @@ func (proxier *Proxier) openPortal(service ServicePortName, info *serviceInfo) e
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIPs {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
+	// This indicates that the load balancer is acting as TCP relay, targeting the machine's IP.
+	// We also treat DNS names the same way, because DNS names are used when we have multiple
+	// addresses for the external load balancer (e.g. AWS ELB)
+	// TODO: This is basically a hack.  If we create load balancers asynchronously it'll likely break.
```

Comment:
It won't actually break, it'll just have an invalid rule temporarily until the public IPs are added.

Did you consider the reverse condition, though? i.e. checking if there were any public IPs that were actually DNS names? I'm not sure we want to trigger this if service.Spec.PublicIPs is just completely empty.